sklearn/base.py

```diff
@@ -385,19 +385,8 @@ def __setstate__(self, state):
         except AttributeError:
             self.__dict__.update(state)
 
-    def _more_tags(self):
-        return _DEFAULT_TAGS
-
-    def _get_tags(self):
-        collected_tags = {}
-        for base_class in reversed(inspect.getmro(self.__class__)):
-            if hasattr(base_class, "_more_tags"):
-                # need the if because mixins might not have _more_tags
-                # but might do redundant work in estimators
-                # (i.e. calling more tags on BaseEstimator multiple times)
-                more_tags = base_class._more_tags(self)
-                collected_tags.update(more_tags)
-        return collected_tags
+    def __sklearn_tags__(self):
```

Comment:
@adrinjalali Removing `_get_tags` is a breaking change if developers are using it. Looking through Github for [self._get_tags(](https://github.com/search?q=self._get_tags&type=code), there are some scikit-learn related use cases for `_get_tags` but it's not too much.

@glemaitre Imbalanced-learn uses `_get_tags` in tests: https://github.com/scikit-learn-contrib/imbalanced-learn/blob/2b6269f9aaea5f058606bf318b8bc36150137dd6/imblearn/utils/estimator_checks.py#L101

Given this is a private API, I am okay with breaking it.